### PR TITLE
Sample Migration to the v1.X version of Python OpenAI

### PR DESCRIPTION
**Pull request by hennachng**
## Purpose

* ... Migrating completion files to the newest version

## Does this introduce a breaking change?

```
[ ] Yes
[X] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[X ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
*  Get the code

```
git clone [repo-address]
cd [repo-name]
git checkout [branch-name]
npm install
```

* Test the code

```
```

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: Basic_Samples/Completions/basic_completions_example_sdk.ipynb [hunk 1]
```diff
@@ -1,159 +1,217 @@
...
+      "Requirement already satisfied: certifi in c:\\users\\henna\\documents\\github\\openai\\venv312\\lib\\site-packages (from httpx<1,>=0.23.0->openai) (2024.2.2)\n",
+      "Requirement already satisfied: httpcore==1.* in c:\\users\\henna\\documents\\github\\openai\\venv312\\lib\\site-packages (from httpx<1,>=0.23.0->openai) (1.0.4)\n",
+      "Requirement already satisfied: h11<0.15,>=0.13 in c:\\users\\henna\\documents\\github\\openai\\venv312\\lib\\site-packages (from httpcore==1.*->httpx<1,>=0.23.0->openai) (0.14.0)\n",
+      "Requirement already satisfied: annotated-types>=0.4.0 in c:\\users\\henna\\documents\\github\\openai\\venv312\\lib\\site-packages (from pydantic<3,>=1.9.0->openai) (0.6.0)\n",
+      "Requirement already satisfied: pydantic-core==2.16.3 in c:\\users\\henna\\documents\\github\\openai\\venv312\\lib\\site-packages (from pydantic<3,>=1.9.0->openai) (2.16.3)\n",
+      "Requirement already satisfied: colorama in c:\\users\\henna\\documents\\github\\openai\\venv312\\lib\\site-packages (from tqdm>4->openai) (0.4.6)\n",
+      "Note: you may need to restart the kernel to use updated packages.\n"
+     ]
+    },
```
Comment (kristapratico) [inline Basic_Samples/Completions/basic_completions_example_sdk.ipynb:44]: We don't need to include the output from the `pip install` in the sample. You should be able to right-click next to this output and select something like "clear cell contents"  to remove it.

File: Basic_Samples/Completions/basic_completions_example_sdk.ipynb [hunk 2]
```diff
@@ -1,159 +1,217 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "e1966c51",
+   "metadata": {},
+   "source": [
+    "### Setup Parameters\n",
+    "\n",
+    "\n",
+    "Here we will load the configurations from _config.json_ file to setup deployment name, openai api base, openai api key and openai api version."
```
Comment (kristapratico) [inline Basic_Samples/Completions/basic_completions_example_sdk.ipynb:92]: Let's update this comment to reflect the changes we made in this sample. We're no longer getting the configuration from the config.json - we're reading the environment variables from the dotenv file.

File: Basic_Samples/Completions/basic_completions_example_sdk.ipynb [hunk 3]
```diff
@@ -1,159 +1,217 @@
...
+    "# The base URL for your Azure OpenAI resource. e.g. \"https://<your resource name>.openai.azure.com\"\n",
+    "azure_endpoint = os.environ['AZURE_OPENAI_ENDPOINT']\n",
+    "\n",
+    "# Currently OPENAI API have the following versions available: 2022-12-01\n",
+    "api_version = os.environ['OPENAI_API_VERSION']\n",
+    "\n",
+    "client = AzureOpenAI(\n",
+    "  api_key = api_key,  \n",
+    "  azure_endpoint = azure_endpoint,\n",
+    "  api_version = api_version\n",
```
Comment (kristapratico) [inline Basic_Samples/Completions/basic_completions_example_sdk.ipynb:117]: Small formatting thing, in Python we shouldn't have any spaces between the keyword argument name and the value we're passing into it.

File: Basic_Samples/Completions/basic_completions_example_sdk.ipynb [hunk 4]
```diff
@@ -1,159 +1,217 @@
...
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "!\"\n",
+      "\n",
+      "# def outer_func(original_string):\n",
+      "#     def inner_func(string):\n",
+      "#         nonlocal original\n"
```
Comment (kristapratico) [inline Basic_Samples/Completions/basic_completions_example_sdk.ipynb:135]: This is kind of a silly response from the model. 😆 Which model are you using? Is it `gpt-35-turbo-instruct`?

**Review by kristapratico — COMMENTED** [review]
Looking good 😸 Only a few comments. Once resolved, we can approve and merge!

**Review by kristapratico — APPROVED** [review]
Perfect! Thanks for your updates, I'm going to go ahead and approve the PR and merge it!

**Comment by kristapratico** [comment]
@hennachng  - actually real quick, can you sign the CLA as described in this [post](https://github.com/Azure-Samples/openai/pull/104#issuecomment-1982150045)?

Just reply to the comment saying `@microsoft-github-policy-service agree`

**Comment by hennachng** [comment]
@microsoft-github-policy-service agree

**Comment by hennachng** [comment]
@microsoft-github-policy-service agree

**Comment by kristapratico** [comment]
@microsoft-github-policy-service rerun

**Pull request merged**